sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -1000,14 +1003,16 @@ def _commit_block_list_options(
     """
     block_lookup = BlockLookupList(committed=[], uncommitted=[], latest=[])
     for block in block_list:
-        if isinstance(block, BlobBlock):
+        if isinstance(block, BlobBlock) and isinstance(block.state, Enum):
             if block.state.value == 'committed' and block_lookup.committed is not None:
                 block_lookup.committed.append(encode_base64(str(block.id)))
             elif block.state.value == 'uncommitted' and block_lookup.uncommitted is not None:
                 block_lookup.uncommitted.append(encode_base64(str(block.id)))
             elif block_lookup.latest is not None:
                 block_lookup.latest.append(encode_base64(str(block.id)))
         else:
+            if block_lookup.latest is None:
+                raise ValueError("block_lookup.latest was None or malformed.")
```

Comment:
So we create `block_lookup` on line 1004 and pass a `latest` meaning we know it's not `None` so this check seems silly. Can you maybe try casting `block_lookup.latest` to not Optional or something instead of this check?

Reply:
So I removed that `None` check and it takes it just fine. BUT still getting errors on the two above....
```
azure\storage\blob\_blob_client_helpers.py:1007: error: Item "None" of "Optional[List[str]]" has no attribute "append"  [union-attr]
azure\storage\blob\_blob_client_helpers.py:1009: error: Item "None" of "Optional[List[str]]" has no attribute "append"  [union-attr]
```

Reply:
Can you use a cast on `block_lookup.latest` to cast it to not Optional? If not, ignore them